src/Sextant.UWP/NavigationView.xaml.cs

```diff
@@ -0,0 +1,456 @@
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Animation;
+using Windows.UI.Xaml.Navigation;
+
+namespace Sextant.UWP
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public partial class NavigationView : Page, IView, IEnableLogger
+    {
+        private readonly IScheduler _backgroundScheduler;
+        private readonly IScheduler _mainScheduler;
+        private readonly IViewLocator _viewLocator;
+        private IFullLogger _logger;
+
+        private IViewModel _currentViewModel;
+        private ContentDialog _contentDialog;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NavigationView"/> class.
+        /// </summary>
+        /// <param name="mainScheduler">The main scheduler to scheduler UI tasks on.</param>
+        /// <param name="backgroundScheduler">The background scheduler.</param>
+        /// <param name="viewLocator">The view locator which will find views associated with view models.</param>
+        /// <param name="rootPage">The starting root page.</param>
+        public NavigationView(IScheduler mainScheduler, IScheduler backgroundScheduler, IViewLocator viewLocator, Page rootPage)
+        {
+            InitializeComponent();
+
+            _backgroundScheduler = backgroundScheduler;
+            _mainScheduler = mainScheduler;
+            _viewLocator = viewLocator;
+            _logger = this.Log();
+
+            PagePopped = Observable
+                .FromEvent<NavigatedEventHandler, NavigationEventArgs>(x => mainFrame.Navigated += x, x => mainFrame.Navigated -= x)
+                .Do(args =>
+                {
+                    if (mainFrame.CanGoBack)
+                    {
+                        backButton.Visibility = Visibility.Visible;
+                    }
+                    else
+                    {
+                        backButton.Visibility = Visibility.Collapsed;
+                    }
+                })
+                .Where(ep => ep.NavigationMode == NavigationMode.Back)
+                .Select(ep => (ep.Content as IViewFor).ViewModel as IViewModel)
+                .WhereNotNull();
+
+            // Observable.FromEvent<NavigatedEventHandler, NavigationEventArgs>(x => mainFrame.Navigated += x, x => mainFrame.Navigated -= x)
+            //    .Subscribe(args =>
+            //    {
+            //        if (mainFrame.CanGoBack)
+            //        {
+            //            backButton.Visibility = Visibility.Visible;
+            //        }
+            //        else
+            //        {
+            //            backButton.Visibility = Visibility.Collapsed;
+            //        }
+            //    });
+            var rootPag = rootPage;
+
+            throw new NotImplementedException("Need to figure out how to add an instanced page to Frame");
```

Comment:
The alternative would be to try to subclass `Frame` and manipulate the `Content`, `BackStack`, etc. through an overload of `NavigateTo` that accepts instances of views.  I don't mind removing this overload because I don't need it.  I guess this only gets used in the native project, not really the cross-platform code so it should be ok.